src/index.ts

```diff
@@ -161,32 +175,26 @@ export function transpose(
 	return commandsToDockerfile(outCommands);
 }
 
-// FIXME: This is taken from resin-io-modules/resin-bundle-resolve
-// export this code to a shared module and import it in this project
-// and resin-bundle-resolve
-export function normalizeTarEntry(name: string): string {
-	const normalized = path.normalize(name);
-	if (path.isAbsolute(normalized)) {
-		return normalized.substr(normalized.indexOf('/') + 1);
-	}
-	return normalized;
-}
-
 const getTarEntryHandler = (
 	pack: tar.Pack,
 	dockerfileName: string,
 	opts: TransposeOptions,
 ) => {
+	const streamToPromise = require('stream-to-promise');
 	return (
-		header: tar.TarHeader,
+		header: tar.Headers,
 		stream: NodeJS.ReadableStream,
 		next: (err?: Error) => void,
 	) => {
 		streamToPromise(stream).then((buffer: Buffer) => {
-			if (normalizeTarEntry(header.name) === dockerfileName) {
+			const name = normalizeTarEntry(header.name);
+			if (name === dockerfileName) {
 				const newDockerfile = transpose(buffer.toString(), opts);
 				pack.entry({ name: 'Dockerfile' }, newDockerfile);
 			} else {
+				if (name === opts.hostQemuPath && opts.qemuFileMode) {
+					header.mode = opts.qemuFileMode;
```

Comment:
This is the change that fixes the "permission denied" error when running `balena build` on Windows when using a NUC computer's Docker daemon (x86_64 balenaEngine) and an ARM app base image. The CLI passes a value for `opts.qemuFileMode`.